ext/json/ext/generator/generator.c

```diff
@@ -189,63 +242,142 @@ static inline void convert_UTF8_to_JSON(FBuffer *out_buffer, VALUE str, const un
 
     unsigned long beg = 0, pos = 0;
 
-#define FLUSH_POS(bytes) if (pos > beg) { fbuffer_append(out_buffer, &ptr[beg], pos - beg); } pos += bytes; beg = pos;
+#ifdef HAVE_ARM_NEON_H
+    const uint8x16_t lower_bound     = vdupq_n_u8(32); 
+    const uint8x16_t backslash       = vdupq_n_u8(92);
+    const uint8x16_t dblquote        = vdupq_n_u8(34);
```

Comment:
```suggestion
    const uint8x16_t lower_bound     = vdupq_n_u8(' '); 
    const uint8x16_t backslash       = vdupq_n_u8('\\');
    const uint8x16_t dblquote        = vdupq_n_u8('\"');
```
Presumably we can use characters here.